server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -758,11 +763,12 @@ private boolean verifyManifestAndClusterState(ClusterMetadataManifest manifest,
             return true;
         }
 
-        private boolean shouldWriteFullClusterState(ClusterState clusterState) {
+        private boolean shouldWriteFullClusterState(ClusterState clusterState, int codecVersion) {
+            assert lastAcceptedManifest == null || lastAcceptedManifest.getCodecVersion() <= codecVersion;
             if (lastAcceptedState == null
                 || lastAcceptedManifest == null
                 || lastAcceptedState.term() != clusterState.term()
-                || lastAcceptedManifest.getOpensearchVersion() != Version.CURRENT) {
```

Comment:
@soosinha I am wondering should we still keep OpenSearchVersion check along with codec as well

Reply:
I will add the version check back as this might provide additional safeguard by handling cases where here is change in the cluster state objects in newer version but the codec version is not updated.